src/code/audio/audio_data.c

```diff
@@ -857,22 +857,99 @@ EnvelopePoint gDefaultEnvelope[] = {
     { ADSR_DISABLE, 0 },
 };
 
-NoteSubEu gZeroNoteSub = { 0 };
-
-NoteSubEu gDefaultNoteSub = {
-    { 1, 1, 0, 0, 0, 0, 0, 0 }, { 0 }, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+NoteSampleState gZeroedSampleState = { 0 };
+
+NoteSampleState gDefaultSampleState = {
+    { true, true, false, false, false, false, false, false },
+    { 0 },
+    0,    // gain
+    0,    // haasEffectLeftDelaySize
+    0,    // haasEffectRightDelaySize
+    0,    // targetReverbVol
+    0,    // harmonicIndexCurAndPrev
+    0,    // combFilterSize
+    0,    // targetVolLeft
+    0,    // targetVolRight
+    0,    // frequencyFixedPoint
+    0,    // combFilterGain
+    NULL, // tunedSample
+    NULL, // filter
+    0,    // unk_18
+    0,    // surroundEffectIndex
+    0,    // unk_1A
 };
 
-u16 gHeadsetPanQuantization[64] = {
-    60, 58, 56, 54, 52, 50, 48, 46, 44, 42, 40, 38, 36, 34, 32, 30, 28, 26, 24, 22, 20, 18,
-    16, 14, 12, 10, 8,  6,  4,  2,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
-    0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
+u16 gHaasEffectDelaySize[64] = {
+    30 * SAMPLE_SIZE,
```

Comment:
Would a macro be useful here, or is this the only place it would be wanted? I'm just looking at the 0s and thinking it'd be nice if they were in the same form, but `0 * SAMPLE_SIZE` seems a bit unnecessary.

Reply:
There's a lot of places in `audio_synthesis` that converts "sample count" (number of samples) to "sample size" (bytes of samples), and they always scale by a factor of two because that is the number of bytes per `s16`. The reason I introduced this is because without it, there were `* 2` all over synthesis, you can see this PR that isolates this specific topic: https://github.com/zeldaret/oot/pull/1289

An alternative to just a `SAMPLE_SIZE` macro would be to replace it with `#define SAMPLE_COUNT_TO_SIZE(numSamples)  ((numSamples) * sizeof(s16))`, but then you would also need:
SAMPLE_COUNT_TO_SIZE32(numSamples)  ((numSamples) * (s32)sizeof(s16))`,
and it gets messy fast.

Reply:
Mm, tricky, don't feel like I have a good solution really. Well, it's a bit like ARRAY_COUNT.